File: javascript/350.IntersectionofTwoArraysII.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersect = function(nums1, nums2) {
    var nums = [];
    for(var i=0;i<nums1.length;i++){
        var element = nums1[i];
        n=nums2.indexOf(element);
        if (n>=0){
            nums.push(element);
            nums1.splice(i, 1);
            nums2.splice(n,1);
            i--;
        }
    }
    return nums;
};

File: python/350.IntersectionofTwoArraysII.py
class Solution(object):
    def intersect(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        map = {}
        for x in nums1:
            if x not in map:
                map[x] = 1
            else:
                map[x] += 1

        res = []
        for y in nums2:
            if y in map and map[y] > 0:
                res.append(y)
                map[y] -= 1

        return res
